core/src/com/unciv/logic/GameStarter.kt

```diff
@@ -585,8 +586,14 @@ object GameStarter {
             val presetStartingLocation = tileMap.startingLocationsByNation[civ.civName]?.randomOrNull()
             val startingLocation = if (presetStartingLocation != null) presetStartingLocation
             else {
-                if (freeTiles.isEmpty()) break // we failed to get all the starting tiles with this minimum distance
-                getOneStartingLocation(civ, tileMap, freeTiles, startScores)
+                val presetAnyStartingLocation = tileMap.startingLocationsByNation[Constants.spectator]?.randomOrNull()
+                if (presetAnyStartingLocation != null) { // Since we don't have any nation specific starting locations, pick one of the free ones to use
+                    tileMap.startingLocationsByNation[Constants.spectator]?.remove(presetAnyStartingLocation)
+                    presetAnyStartingLocation
+                } else {
+                    if (freeTiles.isEmpty()) break // We failed to get all the starting tiles with this minimum distance
```

Comment:
Too much nesting, we need to give a way to better organize this tree of conditions